packages/components/pie-button/src/button.scss

```diff
@@ -304,3 +304,12 @@
     height: var(--btn-icon-size);
     width: var(--btn-icon-size);
 }
+
+// Ensure that any component context that sets hasStackedActions="true"
+// makes the button full width on narrow viewports
+:host-context([hasstackedactions]) .o-btn {
+    // TODO: Move breakpoint sizes into shared CSS component utilties
+    @media (max-width: 768px) {
+        inline-size: 100%;
+    }
+}
```

Comment:
just curious: do we have a strict opinion about overriding component styles this way?

I believe our components should be flexible enough to achieve these cases without relying on custom styles when possible. For example, the button component already supports `isFullWidth`, and setting this behavior through the prop ensures that our components only change from the source and avoid any cascading/overrides.

so in theory, we could do something like 

```
<pie-button isFullWIdth={hasStackActions}></pie-button> 

```
and wrap the footer with flex columns on narrow screens

```
.hasStackedActions footer {
display:flex;
flex-direction: column
}
```

Reply:
I think this is going to become a more prevalent issue as we add more components where their prop-driven styling is screen-size dependent. I.e. A button needs it's full-width prop to only be true on smaller screens.

@ashleynolan @raoufswe I think this would be an excellent candidate for a GitHub discussion, would either of you be happy to set one up?

Reply:
I took a look and I believe we can achieve the desired behavior by making the footer buttons fluid (width 100% of their parent container) and changing the direction of the footer flex container to be `column` if hasStackedAction is true for narrow screens. 


https://github.com/justeattakeaway/pie/assets/28605803/fcf03304-9cf6-4692-8999-69ec6f0a481c


I reverted the changes made to the button locally and added the following: 

```
  &[hasstackedactions] .c-modal-footer  {
        @media (max-width: 768px) {
          flex-direction: column;
        }
    }
```

and added isFullWidth="true" to both footer actions. 

this will ensure that the flex children (footer actions) will expand to span the full width of the parent container along the column direction, and still only span the space they need by default if its set to row, (currently, the default value of footer is (row-reverse).  do you reckon this would work?

Reply:
I'll try it out on Monday and test if it works with those changes across different browsers/widths 👍

Reply:
This works perfectly 🚀  I've just pushed those updates 👍